dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/core/process_definition.py

```diff
@@ -113,6 +126,8 @@ def __init__(
         self._project = project
         self._tenant = tenant
         self.worker_group = worker_group
+        self.warning_type = warning_type
```

Comment:
If we read the value from `configration` file, we better check the input value before we assigned to `self.warning_type`. BTW, better and tests for this
```suggestion
        if warning_type.strip().upper() not in ("FAILURE", "SUCCESS", "ALL", "NONE"):
            raise PyDSParamException("Parameter `warning_type` with unexpect value `%s`", warning_type)
        else:
            self.warning_type = warning_type.strip().upper()
```

Reply:
And then we could remove class `WarningType` in this file